src/experimentation-opportunities/experimentation-opportunities.js

```diff
@@ -11,17 +11,251 @@
+    // await updateRecommendations(oppty, context, site);
+    // update the oppty in the opporrtunities list
+    const index = opportunites.findIndex(
+      (opp) => opp.page === oppty.page && opp.type === oppty.type,
+    );
+    if (index !== -1) {
+      // eslint-disable-next-line no-param-reassign
+      opportunites[index] = oppty;
+    }
+  }
+}
+
+async function createOrUpdateOpportunityEntity(opportunity, context, existingOpportunities) {
+  const { log, dataAccess } = context;
+  const { Opportunity } = dataAccess;
+  const existingOpportunity = existingOpportunities.find(
+    (oppty) => (oppty.getType() === opportunity.type) && oppty.getData()
+    && (oppty.getData().page === opportunity.data.page),
+  );
+  if (existingOpportunity) {
+    if (existingOpportunity.getStatus() === 'NEW') {
+      // remove and create a new opportunity entity with new data
+      log.info(`[${opportunity.type}] Opportunity entity with status: NEW for ${opportunity.data.page} exists, so removing it and creating a new opportunity entity`);
+      await existingOpportunity.remove();
+    } else {
+      log.info(`[${opportunity.type}] Opportunity entity already exists with status: ${existingOpportunity.getStatus()} for ${opportunity.data.page}, so skipping`);
+      return false;
+    }
+  }
+  await Opportunity.create(opportunity);
+  return true;
+}
+
+function convertToOpportunityEntity(oppty, auditData) {
+  return {
+    siteId: auditData.siteId,
+    auditId: auditData.id,
+    runbook: 'https://adobe.sharepoint.com/:w:/r/sites/aemsites-engineering/_layouts/15/Doc.aspx?sourcedoc=%7B19613D9B-93D4-4112-B7C8-DBE0D9DCC55B%7D&file=Experience_Success_Studio_High_Organic_Traffic_Low_CTR_Runbook.docx&action=default&mobileredirect=true',
+    type: 'high-organic-low-ctr',
+    origin: 'AUTOMATION',
+    title: 'page with high organic traffic but low click through rate detected',
+    description: 'Adjusting the wording, images and/or layout on the page to resonate more with a specific audience should increase the overall engagement on the page and ultimately bump conversion.',
+    status: 'NEW',
+    guidance: {
+      recommendations: oppty.recommendations,
+    },
+    tags: ['Engagement'],
+    data: {
+      page: oppty.page,
+      pageViews: oppty.pageViews,
+      samples: oppty.samples,
+      screenshot: oppty.screenshot,
+      trackedKPISiteAverage: oppty.trackedKPISiteAverage,
+      trackedPageKPIName: oppty.trackedPageKPIName,
+      trackedPageKPIValue: oppty.trackedPageKPIValue,
+      opportunityImpact: oppty.opportunityImpact,
+      metrics: oppty.metrics,
+    },
+  };
+}
+
+export async function postProcessor(auditUrl, auditData, context) {
+  const { log } = context;
+  const { dataAccess } = context;
+  let updatedEntities = 0;
+  log.info(`Experimentation Opportunities post processing started for ${auditUrl} from audit ${auditData.id}`);
+  const existingOpportunities = await dataAccess.Opportunity.allBySiteId(auditData.siteId);
```

Comment:
@rpapani does this require error handling as done here:
https://github.com/adobe/spacecat-audit-worker/pull/475/files#diff-212dd58956c947938959c789cc3912af8f1b884de5543486166247e12f8ef215R153